### PR TITLE
Format Attribute Expression

### PR DESCRIPTION
**Pull request by MichaReiser**
## Summary

This PR adds basic formatting for attribute expressions (`a.b`).

## Test Plan

I added a file with new attribute tests (especially around comments)

### CODE REVIEW

**Comment by MichaReiser** [comment]
Current dependencies on/for this PR:
* main
  * **PR #5259** <a href="https://app.graphite.dev/github/pr/astral-sh/ruff/5259" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/astral-sh/ruff/5259?utm_source=stack-comment).

File: crates/ruff_python_formatter/src/expression/expr_attribute.rs [hunk 1]
```diff
@@ -15,28 +15,57 @@ impl FormatNodeRule<ExprAttribute> for FormatExprAttribute {
...
+
         write!(
             f,
             [
-                item.value.format(),
-                requires_space.then_some(space()),
                 text("."),
-                not_yet_implemented_custom_text("NOT_IMPLEMENTED_attr")
+                trailing_comments(trailing_dot_comments),
+                (!leading_attribute_comments.is_empty()).then_some(hard_line_break()),
```
Comment (konstin) [inline crates/ruff_python_formatter/src/expression/expr_attribute.rs:54]: oh i didn't realize write would take an option

**Review by konstin — APPROVED** [review]
this will make a lot more real world code formatted

File: crates/ruff_python_formatter/src/expression/expr_attribute.rs (hunk 1 above)
Comment (MichaReiser) [reply crates/ruff_python_formatter/src/expression/expr_attribute.rs:54]: You can also pass an empty tuple. Don't know when that's useful but it's implemented ;)

**Pull request merged**